### DISCUSSION TITLE
Intellisense VSCode for Flutter Linux project, it doesnt recognize g_autoptr macro

### DISCUSSION THREAD

**Discussion by muizidn** [microsoft/vscode-cpptools#10380] [General]
I have a Flutter Linux project, it uses g_autoptr macro everywhere. Everyone can reproduce the issue by creating a new flutter project for linux platform and check main.cc in linux directory.

I open the project with VSCode and configure it with CMakeTools > Configure.

Intellisesense works fine except that it shows red squiggles for g_autoptrmacro. When I jump to the source code, it greyed out. Even though it seems error in VSCode, it compiles successfully using flutter build linux

I have no experience with C/C++. Hence my assumption is that Intellisense doesn't pass the same definition as the build or something that result in g_autoptrunrecognized or greyed out.

![Screenshot from 2023-01-11 13-45-43](https://user-images.githubusercontent.com/19355715/212078304-7189ca6b-d564-4a80-8264-3428205c012f.png)
![Screenshot from 2023-01-11 13-47-29](https://user-images.githubusercontent.com/19355715/212078312-4a821b4b-1097-4cb4-85a1-cd93ea1ca024.png)
![Screenshot from 2023-01-11 13-48-12](https://user-images.githubusercontent.com/19355715/212078318-0dc3128d-aaf8-4710-9106-91f4abec8d8e.png)

**Comment by bobbrow** [comment]
Looks like a duplicate of https://github.com/microsoft/vscode-cmake-tools/discussions/2940.  We will address it from the CMake angle first. If it turns out we are sending the correct configuration and still getting squiggles, we can open a cpptools issue.

**Reply by muizidn** [reply]
Yes, agree with that. Thank you!

**Reply by bobbrow** [reply]
It's issue #10267
